code/array/toh-vs-each-vs-Hash[]-map-vs-map-toh-vs-each-with-object-vs-tap-each.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "benchmark/ips"
+
+class People
+  def initialize(first_name, last_name)
+    @first_name = first_name
+    @last_name = last_name
+  end
+
+  def login
+    "#{first_name.downcase}-#{last_name.downcase}"
+  end
+
+  private
+
+  attr_reader :first_name, :last_name
+end
+
+PEOPLE = [People.new('John', 'Doe'), People.new('Lola', 'Lanos')].freeze
+
+def fast
+  PEOPLE.to_h { |person| [person.login, person] }
```

Comment:
To be fair to the other benchmarks, this should also call `dup` on `PEOPLE`, I think.

Reply:
Oh yes @nirvdrum, thanks! It might look suspicious without it.

Reply:
If every other benchmark calls `dup`, you're not isolating your changes to the various "to hash" methods. This one will look faster because it's not copying objects. I suppose that's fair in highly niche cases, but for the most part, I don't think people are calling `to_h` on a frozen constant array.